extensions/azurecore/package.json

```diff
@@ -128,6 +128,105 @@
               "Microsoft Authentication Library"
             ],
             "deprecationMessage": "Warning: ADAL has been deprecated, and is scheduled to be removed in a future release. Please use MSAL (default option) instead."
+          },
+          "azure.providerSettingsJson": {
+            "type": "array",
+            "description": "%config.providerSettingsJson%",
+            "scope": "resource",
+            "items": {
+              "type": "object",
+              "properties":{
+                "name": {
+                  "type": "string",
+                  "default": ""
+                },
+                "settings": {
+                  "type": "object",
+                  "properties": {
+                    "metadata": {
+                      "type": "object",
+                      "properties": {
+                        "displayName": {
+                          "type": "string",
+                          "default": ""
+                        },
+                        "id": {
+                          "type": "string",
+                          "default": ""
+                        },
+                        "endpoints": {
+                          "type": "object",
+                          "properties": {
```

Comment:
Also ensure only "required" endpoints are mandatory here, not everything. Users should be able to skip endpoints they don't need to specify.

Use 'required' attribute on "object" types to specify required properties. e.g. on settings object, specify: 
`"required": ["metadata"]`

Doing so will give users errors when they miss a required property:

<img src="https://github.com/microsoft/azuredatastudio/assets/13396919/f4fc8030-7ce8-4820-9dfd-aad4507b1902" width=400 />

More info: https://code.visualstudio.com/api/references/contribution-points#contributes.taskDefinitions